src/Form/BasicSetup.php

```diff
@@ -205,7 +211,7 @@ public function buildForm(array $form, FormStateInterface $form_state, User $use
           '#type' => 'submit',
           '#value' => t('Cancel'),
           '#limit_validation_errors' => array(),
-          '#submit' => array('tfa_basic_setup_form_submit'),
+          '#submit' => array('::submitForm'),
```

Comment:
It might make sense to write a cancelForm method in this class instead of making submitForm more complex. Can you refactor this?